Mage/src/main/java/mage/abilities/dynamicvalue/common/TargetPermanentPowerCount.java

```diff
@@ -16,12 +16,12 @@ public enum TargetPermanentPowerCount implements DynamicValue {
 
     @Override
     public int calculate(Game game, Ability sourceAbility, Effect effect) {
-        Permanent sourcePermanent = game.getPermanent(sourceAbility.getFirstTarget());
-        if (sourcePermanent == null) {
-            sourcePermanent = (Permanent) game.getLastKnownInformation(sourceAbility.getFirstTarget(), Zone.BATTLEFIELD);
+        Permanent targetPermanent = game.getPermanent(effect.getTargetPointer().getFirst(game, sourceAbility));
+        if (targetPermanent == null) {
+            targetPermanent = (Permanent) game.getLastKnownInformation(effect.getTargetPointer().getFirst(game, sourceAbility), Zone.BATTLEFIELD);
```

Comment:
Do not use targetPointer’s game methods for LKI data search (LKI used for looking for PT or other stats at any time) - it can be failed in some use cases cause it checking actual object existing inside (zcc). Little more details here: #6187. Bug example: if you trigger and blink/remove related permanent before trigger resolve then it can be fizzled or do nothing.

Reply:
TurnedFaceUpAllTriggeredAbility uses fixed target so code from #6187 must works fine.
`((FixedTarget) targetPointer).getTargetedPermanentOrLKIBattlefield(game)`

Or just make sure it works fine on trigger source blinks

Reply:
My bad, I *think* it should be good now. Looks like `getTargetPointer().getFirstTargetPermanentOrLKI(game, source)` has become the preferred method for this.